### PR TITLE
Improve xtensa interrupt latency

### PR DESCRIPTION
**Pull request by bjoernQ**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [ ] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
This applies basically the same changes I already did for RISC-V in #1679

Numbers measured on ESP32-S3:

Before: 18.66 / 13.68
After: 13.20 / 7.79

#### Testing
Used same setup as in the RISC-V PR

### CODE REVIEW

File: esp-hal/src/interrupt/mod.rs [hunk 1]
```diff
@@ -114,3 +116,128 @@ impl InterruptHandler {
...
+impl InterruptStatus {
+    const fn empty() -> Self {
+        InterruptStatus {
+            #[cfg(large_intr_status)]
+            status: [0, 0, 0],
+            #[cfg(very_large_intr_status)]
+            status: [0, 0, 0, 0],
+            #[cfg(not(any(large_intr_status, very_large_intr_status)))]
+            status: [0, 0],
+        }
```
Comment (jessebraham) [inline esp-hal/src/interrupt/mod.rs:144]:
```suggestion
        InterruptStatus {
            status: [0u32; STATUS_WORDS],
        }
```

Comment (jessebraham) [reply esp-hal/src/interrupt/mod.rs:144]: Bit nit-picky, but I think we can simplify this

**Review by jessebraham — APPROVED** [review]
One small comment that you're free to ignore, otherwise LGTM!

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Pull request merged**